flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -578,15 +621,27 @@ public void validateExcept(String... prefixesToSkip) {
                     prefixesToSkip.length > 0, "Prefixes to skip can not be empty.");
             final List<String> prefixesList = Arrays.asList(prefixesToSkip);
             consumedOptionKeys.addAll(
-                    configuration.keySet().stream()
+                    allOptions.keySet().stream()
                             .filter(key -> prefixesList.stream().anyMatch(key::startsWith))
                             .collect(Collectors.toSet()));
             validate();
         }
 
-        /** Returns all options of the catalog. */
+        /** Returns all options currently being consumed by the factory. */
         public ReadableConfig getOptions() {
-            return configuration;
+            return allOptions;
+        }
+    }
+
+    /**
+     * Helper utility for validating all options for a {@link CatalogFactory}.
+     *
+     * @see #createCatalogFactoryHelper(CatalogFactory, CatalogFactory.Context)
+     */
+    public static class CatalogFactoryHelper extends FactoryHelper<CatalogFactory> {
```

Comment:
`@PublicEvolving`?

Reply:
The upper class is already declared `@PublicEvolving`.

Reply:
Ah, I guess that makes sense. I went by the fact that it was annotated previously already – which was done by me. :-)